disk uploader task panics when secret already exists

**What happened**:
When user defines in Pipeline retries attribute for disk uploader task as e.g. example: 
```
    - name: upload-image-to-registries
      timeout: 9h0m0s
      retries: 2
```
the uploader task during second run panics with:

```
W1101 10:28:41.020869 18 client_config.go:659] Neither --kubeconfig nor --master was specified. Using the inClusterConfig. This might not work.
{"level":"info","ts":1730456921.0233357,"caller":"disk-uploader/main.go:39","msg":"Creating a new Secret object...","namespace":"windows-builder","name":"windows-efi-2nmxw"}
panic: secrets "windows-efi-2nmxw" already exists [recovered]
panic: secrets "windows-efi-2nmxw" already exists
goroutine 1 [running]:
github.com/kubevirt/kubevirt-tekton-tasks/modules/shared/pkg/exit.HandleExit()
/src/modules/shared/pkg/exit/exit.go:80 +0xbe
panic({0x1c7b420?, 0xc0006109e0?})
/root/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.22.6.linux-amd64/src/runtime/panic.go:770 +0x132
github.com/kubevirt/kubevirt-tekton-tasks/modules/shared/pkg/exit.ExitOrDieFromError(0x3, {0x20c3c00, 0xc0003fd360}, {0x0, 0x0, 0x11?})
/src/modules/shared/pkg/exit/exit.go:60 +0x134
main.main()
/src/cmd/disk-uploader/main.go:135 +0x385
```
**What you expected to happen**:
The task generates secret with random name

**How to reproduce it (as minimally and precisely as possible)**:
100%

/cc @codingben

--- a/cmd/disk-uploader/main.go
+++ b/cmd/disk-uploader/main.go
@@ -40,32 +40,34 @@ func run(opts parse.CLIOptions, k8sClient kubernetes.Interface, virtClient kubec
 
 	log.Logger().Info("Creating a new Secret object...", zap.String("namespace", namespace), zap.String("name", name))
 
-	if err := secrets.CreateVirtualMachineExportSecret(k8sClient, namespace, name); err != nil {
+	vmExportSecret, err := secrets.CreateVirtualMachineExportSecret(k8sClient, namespace, name)
+	if err != nil {
 		return err
 	}
 
 	log.Logger().Info("Creating a new VirtualMachineExport object...", zap.String("namespace", namespace), zap.String("name", name))
 
-	if err := vmexport.CreateVirtualMachineExport(virtClient, kind, namespace, name); err != nil {
+	vmExport, err := vmexport.CreateVirtualMachineExport(virtClient, kind, namespace, name, vmExportSecret.Name)
+	if err != nil {
 		return err
 	}
 
 	log.Logger().Info("Waiting for VirtualMachineExport status to be ready...")
 
-	if err := vmexport.WaitUntilVirtualMachineExportReady(virtClient, namespace, name); err != nil {
+	if err := vmexport.WaitUntilVirtualMachineExportReady(virtClient, namespace, vmExport.Name); err != nil {
 		return err
 	}
 
 	log.Logger().Info("Getting raw disk URL from the VirtualMachineExport object status...")
 
-	rawDiskUrl, err := vmexport.GetRawDiskUrlFromVolumes(virtClient, namespace, name, volumeName)
+	rawDiskUrl, err := vmexport.GetRawDiskUrlFromVolumes(virtClient, namespace, vmExport.Name, volumeName)
 	if err != nil {
 		return err
 	}
 
 	log.Logger().Info("Creating TLS certificate file from the VirtualMachineExport object status...")
 
-	certificateData, err := certificate.GetCertificateFromVirtualMachineExport(virtClient, namespace, name)
+	certificateData, err := certificate.GetCertificateFromVirtualMachineExport(virtClient, namespace, vmExport.Name)
 	if err != nil {
 		return err
 	}
@@ -76,7 +78,7 @@ func run(opts parse.CLIOptions, k8sClient kubernetes.Interface, virtClient kubec
 
 	log.Logger().Info("Getting export token from the Secret object...")
 
-	kvExportToken, err := secrets.GetTokenFromVirtualMachineExportSecret(virtClient, namespace, name)
+	kvExportToken, err := secrets.GetTokenFromVirtualMachineExportSecret(virtClient, namespace, vmExportSecret.Name)
 	if err != nil {
 		return err
 	}

--- a/modules/disk-uploader/pkg/secrets/secrets.go
+++ b/modules/disk-uploader/pkg/secrets/secrets.go
@@ -13,17 +13,17 @@ import (
 	"k8s.io/client-go/kubernetes"
 )
 
-func CreateVirtualMachineExportSecret(k8sClient kubernetes.Interface, namespace, name string) error {
+func CreateVirtualMachineExportSecret(k8sClient kubernetes.Interface, namespace, baseName string) (*corev1.Secret, error) {
 	length := 20
 	token, err := generateSecureRandomString(length)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
 	v1Secret := &corev1.Secret{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      name,
-			Namespace: namespace,
+			GenerateName: baseName + "-",
+			Namespace:    namespace,
 		},
 		StringData: map[string]string{
 			"token": token,
@@ -31,11 +31,10 @@ func CreateVirtualMachineExportSecret(k8sClient kubernetes.Interface, namespace,
 	}
 
 	if err := ownerreference.SetPodOwnerReference(k8sClient, v1Secret); err != nil {
-		return err
+		return nil, err
 	}
 
-	_, err = k8sClient.CoreV1().Secrets(namespace).Create(context.Background(), v1Secret, metav1.CreateOptions{})
-	return err
+	return k8sClient.CoreV1().Secrets(namespace).Create(context.Background(), v1Secret, metav1.CreateOptions{})
 }
 
 func GetTokenFromVirtualMachineExportSecret(k8sClient kubernetes.Interface, namespace, name string) (string, error) {

--- a/modules/disk-uploader/pkg/secrets/secrets_test.go
+++ b/modules/disk-uploader/pkg/secrets/secrets_test.go
@@ -50,12 +50,12 @@ var _ = Describe("VMExport", func() {
 			)
 			Expect(err).NotTo(HaveOccurred())
 
-			err = secrets.CreateVirtualMachineExportSecret(kubeClient, namespace, name)
+			_, err = secrets.CreateVirtualMachineExportSecret(kubeClient, namespace, name)
 			Expect(err).NotTo(HaveOccurred())
 		})
 
 		It("should return error when set pod owner reference failed", func() {
-			err := secrets.CreateVirtualMachineExportSecret(kubeClient, namespace, name)
+			_, err := secrets.CreateVirtualMachineExportSecret(kubeClient, namespace, name)
 			Expect(err).To(MatchError(errors.IsNotFound, "errors.IsNotFound"))
 		})
 	})

--- a/modules/disk-uploader/pkg/vmexport/vmexport.go
+++ b/modules/disk-uploader/pkg/vmexport/vmexport.go
@@ -24,29 +24,28 @@ const (
 	sourcePVC        string = "pvc"
 )
 
-func CreateVirtualMachineExport(virtClient kubecli.KubevirtClient, exportSourceKind, exportSourceNamespace, exportSourceName string) error {
-	source, err := getExportSource(exportSourceKind, exportSourceName)
+func CreateVirtualMachineExport(virtClient kubecli.KubevirtClient, exportSourceKind, exportSourceNamespace, baseExportSourceName, secretName string) (*v1beta1.VirtualMachineExport, error) {
+	source, err := getExportSource(exportSourceKind, baseExportSourceName)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
 	v1VmExport := &v1beta1.VirtualMachineExport{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      exportSourceName,
-			Namespace: exportSourceNamespace,
+			GenerateName: baseExportSourceName + "-",
+			Namespace:    exportSourceNamespace,
 		},
 		Spec: v1beta1.VirtualMachineExportSpec{
-			TokenSecretRef: &exportSourceName,
+			TokenSecretRef: &secretName,
 			Source:         source,
 		},
 	}
 
 	if err := ownerreference.SetPodOwnerReference(virtClient, v1VmExport); err != nil {
-		return err
+		return nil, err
 	}
 
-	_, err = virtClient.VirtualMachineExport(exportSourceNamespace).Create(context.Background(), v1VmExport, metav1.CreateOptions{})
-	return err
+	return virtClient.VirtualMachineExport(exportSourceNamespace).Create(context.Background(), v1VmExport, metav1.CreateOptions{})
 }
 
 func WaitUntilVirtualMachineExportReady(client kubecli.KubevirtClient, namespace, name string) error {

--- a/modules/disk-uploader/pkg/vmexport/vmexport_test.go
+++ b/modules/disk-uploader/pkg/vmexport/vmexport_test.go
@@ -73,7 +73,7 @@ var _ = Describe("VMExport", func() {
 				)
 				Expect(err).NotTo(HaveOccurred())
 
-				err = vmexport.CreateVirtualMachineExport(virtClient, resource, namespace, name)
+				_, err = vmexport.CreateVirtualMachineExport(virtClient, resource, namespace, name, "fake-secret")
 				Expect(err).NotTo(HaveOccurred())
 				Expect(errors.IsNotFound(err)).To(BeFalse())
 			},
@@ -83,12 +83,12 @@ var _ = Describe("VMExport", func() {
 		)
 
 		It("should return error when export-source-kind invalid", func() {
-			err := vmexport.CreateVirtualMachineExport(virtClient, "fake", namespace, name)
+			_, err := vmexport.CreateVirtualMachineExport(virtClient, "fake", namespace, name, "fake-secret")
 			Expect(err).To(MatchError("invalid export-source-kind: fake, must be one of vm, vmsnapshot, pvc"))
 		})
 
 		It("should return error when set pod owner reference failed", func() {
-			err := vmexport.CreateVirtualMachineExport(virtClient, "vm", namespace, name)
+			_, err := vmexport.CreateVirtualMachineExport(virtClient, "vm", namespace, name, "fake-secret")
 			Expect(err).To(MatchError(errors.IsNotFound, "errors.IsNotFound"))
 		})
 	})

